packages/sit-onyx/src/types/utils.ts

```diff
@@ -2,3 +2,11 @@
  * Recursive / deep implementation of TypeScript's built-in `Partial<T>` type.
  */
 export type DeepPartial<T> = T extends object ? { [P in keyof T]?: DeepPartial<T[P]> } : T;
+
+/**
+ * Recursive type to allow passing additional properties to a given object.
+ * It doesn't alter the original type but allows nested objects to have additional properties not defined in the original type.
+ */
+export type DeepExtendable<T> = T extends object
```

Comment:
Because while the existing "DeepPartial" type makes all properties of an object optional, it won't allow for other properties. That's also why a type assertion was needed within all the tests for i18n. During my tests when I didn't overwrite the t function this caused multiple type issues when trying to pass my own messages. By now it should no longer be neccessary but I decided to leave it in there to restore duck typing for the messages passed to createOnyx. But if you want we I can also remove it but that issue might come up in the future again.

Reply:
If its not needed, I'd like to remove it since I don't see why the onyx translations should be extended because they are only used internally and adding new translations to onyx that are not used by any components does not make sense to me :)